solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1703,7 +1703,8 @@ public void publish(final CoreDescriptor cd, final Replica.State state, boolean
       DocCollection coll = zkStateReader.getCollection(collection);
       if (forcePublish || sendToOverseer(coll, coreNodeName)) {
         overseerJobQueue.offer(Utils.toJSON(m));
-      } else {
+      }
```

Comment:
@noblepaul what is the motivation to remove `else` part here? to publish the prs state of replica?

Q: solrcore will update the replica status only? if so then for prs collection we don't need to update the state.json file.